src/storage_manager.rs

```diff
@@ -56,7 +56,12 @@ pub trait StorageManagerInterface: Send + Sync {
     ///
     /// Callback function pointer to native (kotlin/swift) code for
     /// getting a key.
-    fn get(&self, key: Key) -> Result<Option<Value>, StorageManagerError>;
+    fn get(&self, key: Key) -> Result<Value, StorageManagerError>;
```

Comment:
The `StorageManagerError` enum does not have a variant for "not found", and I think it is more idiomatic to separate "not found" from IO errors.
```suggestion
    fn get(&self, key: Key) -> Result<Option<Value>, StorageManagerError>;
```

Reply:
From an API point of view, this does mean the caller has to do two error checks rather than one, and from the caller point of view I don't think the two separate error types are useful.  Personally my inclination would be to add a `StorageManagerError::NotFound` rather than make the caller check (a) if it worked, and then (b) if they actually got anything.

Reply:
I'm not going to die on this hill but my opinion is that an application probably wants to handle these cases differently:
IOError -> probably not recoverable, log the specific error and move on.
NotFound -> probably recoverable, may involve user interaction depending on the context.

Is it really unusual to have a method that can throw and return a nullable value?
> func get(key: Key) throws  -> Value?